src/dcos_e2e/docker_utils/__init__.py

```diff
@@ -0,0 +1,122 @@
+from typing import Any, Dict, Optional, Tuple
+
+import docker
+
+
+class DockerLoopbackVolume():
+    """
+    A loopback device sidecar, created in a Docker container.
+    """
+
+    def __init__(
+        self,
+        size: int,
+        labels: Optional[Dict[str, str]] = None,
+    ) -> None:
+        """
+        Create a loopback device pointing to a block device running
+        in a container. This can be used to provide an unformatted
+        disk to a cluster.
+
+        Args:
+            size: Size of the block device in Megabytes.
+            labels: Docker labels to add to the container.
+        """
+        self.size = size
+        docker_container_labels = labels or {}
+
+        self.container, self.path = \
+            DockerLoopbackVolume.create(size, docker_container_labels)
+
+    @staticmethod
+    def create(
+        size: int,
+        labels: Optional[Dict[str, str]] = None,
+    ) -> Tuple[docker.models.containers.Container, str]:
+        """
+        Create a loopback device pointing to a block device running
+        in a container.
+
+        Args:
+            size: Size of the block device in Megabytes.
+            labels: Docker labels to add to the container.
+        Returns:
+            A tuple containing the container and the path of
+            the loopback device.
+        """
+        client = docker.from_env(version='auto')
+
+        container = client.containers.create(
+            name='dcos-e2e-loopback-sidecar',
+            privileged=True,
+            detach=True,
+            tty=True,
+            image='centos:7',
+            labels=labels or {},
+        )
+        container.start()
+
+        create_loopback_device = \
+            'dd if=/dev/zero of=/volume0 bs=1M count={0};'.format(size)
+
+        exit_code, output = container.exec_run(
+            cmd=['/bin/bash', '-c', create_loopback_device],
+        )
+        assert exit_code == 0, output.decode()
+
+        setup_loopback_device = 'losetup --find --show /volume0;'
+
+        exit_code, output = container.exec_run(
+            cmd=['/bin/bash', '-c', setup_loopback_device],
+        )
+        assert exit_code == 0, output.decode()
+
+        path = output.decode().rstrip()
+
+        write_device_path = 'echo {0} > loopback_device_path'.format(path)
+
+        exit_code, output = container.exec_run(
+            cmd=['/bin/bash', '-c', write_device_path],
+        )
+        assert exit_code == 0, output.decode()
+
+        return (container, path)
+
+    @staticmethod
+    def destroy(container: docker.models.containers.Container) -> None:
+        """
+        Destroy a container containing a block device that is used with
```

Comment:
The function does not seem to destroy a container like the docstring says.

Reply:
Thanks, I'll update the function to include `container.stop` and `container.remove` calls.